web/packages/teleport/src/Discover/Shared/AwsAccount/AwsAccount.tsx

```diff
@@ -96,36 +124,69 @@ export function AwsAccount() {
       nodesAccess.list &&
       nodesAccess.read; // Needed for TestConnection flow
   }
+  if (isAddingAwsApp) {
+    roleTemplate = integrationAndAppRW;
+    const appAccess = storeUser.getAppServerAccess();
+    hasAccess =
+      hasAccess &&
+      // required to upsert app server
+      appAccess.create &&
+      appAccess.edit &&
+      // required to list and read app servers
+      appAccess.list &&
+      appAccess.read;
+  }
 
-  const { attempt, run } = useAttempt(hasAccess ? 'processing' : '');
-
-  const [awsIntegrations, setAwsIntegrations] = useState<Option[]>([]);
   const [selectedAwsIntegration, setSelectedAwsIntegration] =
     useState<Option>();
 
   useEffect(() => {
-    if (hasAccess) {
-      fetchAwsIntegrations();
+    if (hasAccess && attempt.status === '') {
+      fetch();
+    }
+  }, [attempt.status, fetch, hasAccess]);
+
+  async function fetchAwsIntegrationsWithApps(
```

Comment:
Personally, I'd move the function out of the component, but I will leave it up to you.